src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,40 +205,82 @@ function updateStatsCheckBox() {
     $('input[id=statsShownCheckBox]').val($('#show-stats-checkbox').is(':checked'));
 }
 
-function toggleCollapse(e, pans) {
-    var expand = 'Expand';
-    var collapse = 'Collapse';
-    var panels = pans || $('div.panel-collapse');
-
-    if ($(e).html().trim().startsWith(expand)) {
-        var i = 0;
-        for (var idx = 0; idx < panels.length; idx++) {
-            if ($(panels[idx]).attr('class').indexOf('in') === -1) {
-                setTimeout(showSingleCollapse, 50 * i, panels[idx]);
-                i++;
-            }
-        }
-        var htmlString = $(e).html();
-        htmlString = htmlString.replace(expand, collapse);
-        $(e).html(htmlString);
-        var tooltipString = $(e).attr('data-original-title').replace(expand, collapse);
-        $(e).attr('title', tooltipString).tooltip('fixTitle').tooltip('show');
+function toggleCollapse(element, panels) {
+    var EXPAND = 'Expand';
+    var COLLAPSE = 'Collapse';
+    // {@code panels} is undefined when the method is called by clicking on collapse panels button.
+    var panelsDefined = panels || $('div.panel-collapse');
+    var isElementAnExpandButton = $(element).html().trim().startsWith(EXPAND);
+
+    if (isElementAnExpandButton) {
+        expand(panelsDefined, element);
+        replaceInHtml(element, EXPAND, COLLAPSE);
     } else {
-        var j = 0;
-        for (var k = 0; k < panels.length; k++) {
-            if ($(panels[k]).attr('class').indexOf('in') !== -1) {
-                setTimeout(hideSingleCollapse, 100 * j, panels[k]);
-                j++;
+        collapse(panelsDefined);
+        replaceInHtml(element, COLLAPSE, EXPAND);
+    }
+}
+/**
+ * Expand each panel. If the panel data is not loaded, load it by Ajax first before expansion.
+ */
+function expand(panels, element) {
+    var isElementCollapsePanelsButton = $(element).is($('#collapse-panels-button'));
```

Comment:
Previously this element was recognized as an 'expand button'; what's this about?